bubble_sort.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+# rubocop:disable Metrics/AbcSize
+
+def bubble_sort(arr)
+  swap = true
+  while swap
+    swap = false
+    (arr.length - 1).times do |x|
+      swap = true if arr[x] > arr[x + 1]
+      arr[x], arr[x + 1] = arr[x + 1], arr[x] if arr[x] > arr[x + 1]
+    end
+  end
+  print arr
+end
+
+arr = [1, 4, 5, 66, 7, 3, 12_312, 2]
+
+bubble_sort(arr)
+
+def bubble_sort_by(arr)
+  swap = true
+  while swap
+    swap = false
+    (arr.length - 1).times do |x|
+      b = arr[x + 1]
+      arr[x], arr[x + 1] = arr[x + 1], arr[x] if yield(arr[x], b) >= 0
+      swap = true if yield(arr[x], b) >= 0
+    end
+  end
+  print arr
+end
+
+bubble_sort_by(arr) { |a, b| a - b }
+
+unsorted = %w[hi hello hey]
+puts bubble_sort_by(unsorted) { |left, right| right.length - left.length }
+puts bubble_sort(unsorted)
```

Comment:
- **[REQUIRED]** No debugging instructions in committed code (as discussed [here](https://questions.microverse.org/t/debugging-instructions-code/346)) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/section/debugging_requirement.png) from Pathwright)

Reply:
Debugging instructions on lines 34-38 have been removed (as well as on line 17). Thanks!